packages/flutter/lib/src/foundation/memory_allocations.dart

```diff
@@ -0,0 +1,176 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui' as ui;
+
+///
+abstract class ObjectEvent{}
+
+///
+typedef ObjectEventListener = void Function(ObjectEvent);
+
+///
+class ObjectCreated implements ObjectEvent {
+  ///
+  ObjectCreated({
+    required this.library,
+    required this.klass,
+    required this.object,
+    this.token,
+    this.details = const <Object>[],
+  });
+
+  ///
+  final String library;
+
+  ///
+  final String klass;
+
+  /// Reference to this object should not be stored in any
+  /// long living place as it will prevent garbage collection.
+  final Object object;
+
+  /// A token that uniquely identify object, for object tracking
+  /// accross events.
+  ///
+  /// If not provided, the consumers of the event may use
+  /// [identityHashCode] and handle small risk of duplicates.
+  /// The object's token should be the same accross events.
+  final Object? token;
+
+  ///
+  final List<Object> details;
+}
+
+///
+class ObjectDisposed implements ObjectEvent {
+
+  ///
+  ObjectDisposed({
+    required this.object,
+    this.details = const <Object>[],
+    this.token }
+  );
+
+  ///
+  final Object object;
+
+  ///
+  final Object? token;
+
+  ///
+  final List<Object> details;
+}
+
+
+/// The event contains some tracing information that may help with memory
+/// troubleshooting.
+///
+/// For example, it may be information about ownership transfer
+/// or state change.
+class ObjectTraced implements ObjectEvent {
+
+  ///
+  ObjectTraced(
+    this.object, {
+    this.details = const <Object>[],
+    this.token, }
+  );
+
+  ///
+  final Object object;
+
+  ///
+  final Object? token;
+
+  ///
+  final List<Object> details;
+}
+
+///
+class MemoryAllocations {
+  MemoryAllocations._();
+
+  ///
+  // Lint is ignored here, because 'late' is needed for lazy pattern.
+  // ignore: unnecessary_late
+  static late final MemoryAllocations instance = MemoryAllocations._();
+
+  List<ObjectEventListener>? _listeners;
+
+  ///
+  void addListener(ObjectEventListener listener){
+    if (_listeners == null) {
+      _listeners = <ObjectEventListener>[];
+      _subscribeToSdkObjects();
+    }
+    _listeners!.add(listener);
+  }
+
+  ///
+  void removeListener(ObjectEventListener listener){
+    _listeners?.remove(listener);
+    if (_listeners?.isEmpty ?? false) {
+      _listeners = null;
+      _unSubscribeFromSdkObjects();
+    }
+  }
+
+  ///
+  void registerObjectEvent(ObjectEvent objectEvent) {
```

Comment:
dispatch instead of register?

Reply:
Is this supposed to be public API? OR just for internal use?

Reply:
It should be public and used for Flutter Framework (State, Render, Layer, ChangeNotifier) and custom user objects.